### DISCUSSION TITLE
Experiencing service latency

### DISCUSSION THREAD

**Discussion by rklf** [komoot/photon#836] [General]
Hello,

Thanks for your work. Unfortunately, I'm experiencing latency using the service (photon.komoot.io) am I the only one or you're aware of this?

<img width="1512" alt="image" src="https://github.com/user-attachments/assets/a713caab-5b4a-47ca-ad29-00b02cc0c6ec">

Any idea?

Thanks

**Comment by lonvia** [comment]
That's because somebody flooding the service with requests. I've banned the worst offending IPs now.

Please, people, behave yourselves. This is only just one tiny server we run for the benefit of everybody. Sending hundreds of requests in parallel will accomplish nothing because there still is just one server responding. All this achieves is just that there will be access restrictions sooner or later.

If you have lots of queries to make, getting your own instance of Photon is as easy as downloading two files.

**Reply by rklf** [reply]
Thanks for your answer. Won't you guys be able to implement a fail2ban and a free access key to use the API ? It would restrict at least those flooding requests and if so you'd be able to identify them by its access key.

**Reply by mtmail** [reply]
Who has the time to build a API key management system with user accounts, privacy policy and support for a $0 service? People would use fake/disposable email address to signup (https://github.com/rklf/yopmail), it won't stop those who want to flood.

**Reply by rklf** [reply]
Off-topic — Repo you mentioned isn't aimed to help people creating disposable email address, w/ or w/o this people would still be able to use disposable services...
Very basic API key management isn't a big deal, it would at least retain people who's falling on the API link randomly. Overall main advice was fail2ban though.

**Comment by lonvia** [comment]
As far as I can tell, there is a badly programmed Ruby application out there, which has been hogging up to 60% of server capacity yesterday. Basically, someone thought nothing to send a reverse query to the server every time the internal GPS sends new location information. Only, some GPS send updated locations as often as 10 times per second. 90% of requests were just blocked by nginx with a 404 but that still managed to hog all the sever capacity.

Automatic IP banning is now in place but the service will remain slow to respond until the programmers of that application fix the mess they have created.
